exonum/src/runtime/rust/interfaces.rs

```diff
@@ -0,0 +1,161 @@
+    ///
+    /// This method is called then some external conditions occur and thus this happens
+    /// outside of the transaction execution, which means that errors that occur during the
+    /// execution of this method may be ignored.
+    ///
+    /// # Execution policy
+    ///
+    /// At the moment, this method can only be called on behalf of the supervisor service instance.
+    /// In other words, only a method with the specified [identifier] can call this method.
+    ///
+    /// [identifier]: ../../constant.SUPERVISOR_SERVICE_ID.html
+    fn apply_config(
+        &self,
+        context: TransactionContext,
+        params: Self::Params,
+    ) -> Result<(), ExecutionError>;
+}
+
+impl<T: BinaryValue> Interface for dyn Configure<Params = T> {
+    const INTERFACE_NAME: &'static str = CONFIGURE_INTERFACE_NAME;
+
+    fn dispatch(
+        &self,
+        context: TransactionContext,
+        method: MethodId,
+        payload: &[u8],
+    ) -> Result<(), ExecutionError> {
+        match method {
+            VERIFY_CONFIG_METHOD_ID => self.verify_config(
+                context,
+                T::from_bytes(payload.into()).map_err(DispatcherError::malformed_arguments)?,
+            ),
+
+            APPLY_CONFIG_METHOD_ID => self.apply_config(
+                context,
+                T::from_bytes(payload.into()).map_err(DispatcherError::malformed_arguments)?,
+            ),
+
+            other => {
+                let kind = DispatcherError::NoSuchMethod;
+                let message = format!(
+                    "Method with ID {} is absent in the 'Configure' interface of the instance `{}`",
+                    other, context.instance.name,
+                );
+                Err((kind, message)).map_err(From::from)
+            }
+        }
+    }
+}
+
+/// A helper struct for invoking the [`Configure`] interface methods on the specified service instance.
+///
+/// [`Configure`]: trait.Configure.html
+#[derive(Debug)]
+pub struct ConfigureCall<'a>(CallContext<'a>);
+
+impl<'a> From<CallContext<'a>> for ConfigureCall<'a> {
+    fn from(context: CallContext<'a>) -> Self {
+        Self(context)
+    }
+}
+
+impl<'a> ConfigureCall<'a> {
+    /// Invoke the corresponding [method](trait.Configure.html#tymethod.verify_config)
+    /// of the interface.
+    pub fn verify_config(&self, params: impl BinaryValue) -> Result<(), ExecutionError> {
+        self.0
+            .call(CONFIGURE_INTERFACE_NAME, VERIFY_CONFIG_METHOD_ID, params)
+    }
+
+    /// Invoke the corresponding [method](trait.Configure.html#tymethod.apply_config)
+    /// of the interface.
+    pub fn apply_config(&self, params: impl BinaryValue) -> Result<(), ExecutionError> {
+        self.0
+            .call(CONFIGURE_INTERFACE_NAME, APPLY_CONFIG_METHOD_ID, params)
+    }
+}
+
+/// Verify that the caller of this method is supervisor service.
+pub fn caller_is_supervisor(caller: &Caller, _: &dyn Snapshot) -> Result<(), ExecutionError> {
```

Comment:
`caller_is_supervisor` is expected to return `bool` because of its name. Maybe `verify_caller_is_supervisor`?